src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala

```diff
@@ -93,18 +93,35 @@ class TaskReplaceActor(
       // reset the launch queue delay
       logger.info("Resetting the backoff delay before restarting the runSpec")
       launchQueue.resetDelay(runSpec)
-    }.pipeTo(self)
 
-    // it might be possible, that we come here, but nothing is left to do
-    checkFinished()
+      // it might be possible, that we come here, but nothing is left to do
+      checkFinished()
+
+      Done
+    }.pipeTo(self)
   }
 
   override def postStop(): Unit = {
     eventBus.unsubscribe(self)
     super.postStop()
   }
 
-  override def receive: Receive = readinessBehavior orElse replaceBehavior
+  override def receive: Receive = initializing
+
+  private def initializing: Receive = {
+    case Done =>
```

Comment:
I so don't like how we rely on Done as a msg on so many places. For me it's like sending a message of type Any... (hard to tell where it came from, might be coming from any future)

But that's an overall problem, not necessarily with this particular change :)

Reply:
I was thinking about using a custom message such as `case object Initialized`